R/article.R

```diff
@@ -252,11 +252,23 @@ lipics_article <- function(
 ) {
   # quick dev shortcut for Ubuntu: click "Install and restart" then run:
   # unlink("MyArticle/", recursive = TRUE); rmarkdown::draft("MyArticle.Rmd", template = "lipics", package = "rticles", edit = FALSE); rmarkdown::render("MyArticle/MyArticle.Rmd"); system(paste0("xdg-open ", here::here("MyArticle", "MyArticle.pdf")))
-  pdf_document_format(
+  base <- pdf_document_format(
     "lipics", latex_engine = latex_engine,
     citation_package = citation_package, keep_tex = keep_tex,
     md_extensions = md_extensions, ...
   )
+
+  # we deactivate bookdown addition of environment definitions in preamble
+  # to avoid conflict as lipics already defines some
+  opts <- options(bookdown.theorem.preamble = FALSE)
+  on_exit <- function() options(opts)
+
+  base_on_exit <- base$on_exit
+  base$on_exit <- function() {
+    if (is.function(base_on_exit)) base_on_exit()
+    on_exit()
+  }
+  base
```

Comment:
That sounds too complicated. I'd just add a single line of `options(bookdown.theorem.preamble = FALSE)` to `skeleton.Rmd` (e.g., in the `setup` chunk).

No strong opinion, though. You can decide by yourself.

Reply:
Oh ok. I recognize this is fairly complicated - It took me some time to write this correctly :)

but I did not want to expose this "kind of internal" option to users. I also choose to put that inside the output function so that previous articles with old Rmd template would benefit from it when using new bookdown article. 

Setting on exit hook to `render()` is coming more and more in our usage. Currently doing this through the format is the only way I know to do it properly. Maybe we are missing helper function to that properly. `rmarkdown:::merge_on_exit()` is the one that would simplify currently 

```r
opts <- options(bookdown.theorem.preamble = FALSE)
base$on_exit <- rmarkdown:::merge_on_exit(base$on_exit, function() options(opts))
```

There may be other way. 

Anyway, I am fine also by exposing this in a setup chunk for the articles skeleton.

Reply:
I decided to follow you advice on this one and simplify this. Thanks !